kobo/apps/mass_emails/tasks.py

```diff
@@ -33,20 +67,130 @@ def mark_old_enqueued_mass_email_record_as_failed():
     )
 
 
-def enqueue_mass_email_records(email_config):
-    """
-    Creates a email job and enqueues email records for users based on query
-    """
-    job = MassEmailJob.objects.create(email_config=email_config)
-    users = USER_QUERIES.get(email_config.query, lambda: [])()
+def render_template(template, data):
+    rendered = template
+    for placeholder, value in templates_placeholders.items():
+        rendered = rendered.replace(placeholder, data[value])
+    return rendered
 
-    records = [
-        MassEmailRecord(user=user, email_job=job, status=EmailStatus.ENQUEUED)
-        for user in users
-    ]
-    MassEmailRecord.objects.bulk_create(records)
 
-    logging.info(
-        f'Created {len(records)} MassEmailRecord(s) for {email_config.name} '
-        f'with query {email_config.query}'
-    )
+class MassEmailSender:
+
+    def __init__(self):
+        self.today = localdate()
+        self.cache_key_prefix = f'mass_emails_{self.today.isoformat()}_email_limits'
```

Comment:
What do you think about changing this to `emails_remaining` instead of `email_limits`? By no means mandatory but I think it would be a little clearer.